cmd/horcrux/cmd/key2shares.go

```diff
@@ -17,32 +17,66 @@ package cmd
 
 import (
 	"fmt"
+	"os"
+	"path/filepath"
 	"strconv"
 
-	"github.com/cometbft/cometbft/libs/os"
 	"github.com/spf13/cobra"
 	"github.com/strangelove-ventures/horcrux/signer"
 )
 
+func createCosignerDirectoryIfNecessary(id int) (string, error) {
+	dir := fmt.Sprintf("cosigner_%d", id)
+	dirStat, err := os.Stat(dir)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return "", fmt.Errorf("unexpected error fetching info for cosigner directory: %w", err)
+		}
+		if err := os.Mkdir(dir, 0700); err != nil {
+			return "", fmt.Errorf("failed to make directory for cosigner files: %w", err)
+		}
+		return dir, nil
+	}
+	if !dirStat.IsDir() {
+		return "", fmt.Errorf("path must be a directory: %s", dir)
+	}
+	return dir, nil
+}
+
 // CreateCosignerSharesCmd is a cobra command for creating cosigner shares from a priv validator
-func createCosignerSharesCmd() *cobra.Command {
+func createCosignerEd25519SharesCmd() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:     "create-shares [priv_validator.json] [threshold] [shares]",
+		Use:     "create-ed25519-shares [chain-id] [priv_validator.json] [threshold] [shares]",
 		Aliases: []string{"shard", "shares"},
-		Args:    validateCreateCosignerShares,
+		Args:    cobra.ExactArgs(4),
```

Comment:
\> 1 or 2 positional args gets confusing, especially since at the end of the day they are all strings. 

Consider using required flags instead. I think that improves U/X. Personally, when calling this command I'm always afraid I mix up the threshold and the shares. Flags make it explicit. 

```
--chain-id
--key-file
--threshold
--shares
```

Reply:
As a bonus, with cobra, you don't have to do the string conversion if they are flags.

Reply:
Can be a future PR if you agree.

Reply:
good UX gain. updated to use the flags